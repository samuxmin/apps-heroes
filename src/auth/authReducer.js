import { types } from "../types/types";

// const state = {
//     name:'Samuel',
//     logged:false
// }

export const authReducer = (state = {}, action) => {
   
    switch (action.type) {
        case types.login:
            return{
                ...action.payload, 
                logged:true
            }
        case types.logout:
            return{
                logged:false
            }

        default:
            return state;
    }
}
