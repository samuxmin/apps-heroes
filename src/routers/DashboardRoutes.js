import React from 'react'
import { Switch,Route, Redirect } from 'react-router-dom'
import { DcScreen } from '../component/dc/DcScreen'
import { HeroScreen } from '../component/heroes/HeroScreen'
import { AllHeroes } from '../component/main/AllHeroes'
import { MarvelScreen } from '../component/marvel/MarvelScreen'
import { SearchScreen } from '../component/search/SearchScreen'
import { Navbar } from '../component/ui/Navbar'

export const DashboardRoutes = () => {
    return (
        <>
            <Navbar/>
            <div className='app'>
                <Switch>
                    <Route exact path='/' component={AllHeroes}/>
                    <Route exact path='/search' component={SearchScreen} />
                    <Route exact path="/marvel" component={ MarvelScreen }/>
                    <Route exact path="/hero/:heroeId" component={ HeroScreen }/>
                    <Route exact path="/dc" component={ DcScreen }/>

                    <Redirect to='/marvel' />
                </Switch>
            </div>
        </>
    )
}
