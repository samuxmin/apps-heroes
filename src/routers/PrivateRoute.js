import React from 'react';
import { Route,Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';


export const PrivateRoute = ({isAuth, component:Component,...rest
    }) => {
    const lastPath = rest.location.pathname + rest.location.search || rest.location.pathname
    localStorage.setItem('lastPath', lastPath);

    return (
        <Route {...rest} 
            component={(props)=>(
                (isAuth)
                    ?(<Component {...props}/>)
                    :(<Redirect to='/login' />)

                )}
            />
        )
}
PrivateRoute.propTypes = {
    isAuth: PropTypes.bool.isRequired,
    component: PropTypes.func.isRequired
}