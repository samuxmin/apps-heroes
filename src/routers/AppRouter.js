import React, { useContext } from "react";
import { BrowserRouter as Router, Switch } from "react-router-dom";
import { LoginScreen } from "../component/login/LoginScreen";
import { PrivateRoute } from "./PrivateRoute";
import { AuthContext } from "../auth/AuthContext";
import { DashboardRoutes } from "./DashboardRoutes";
import { PublicRoute } from "./PublicRoute";
export const AppRouter = () => {
  const { user } = useContext(AuthContext);

  return (
    <Router>
      <div>
        <Switch>
          <PublicRoute
            exact
            path="/login"
            component={LoginScreen}
            isAuth={user.logged}
          />
          <PrivateRoute
            isAuth={user.logged}
            path="/"
            component={DashboardRoutes}
          />
        </Switch>
      </div>
    </Router>
  );
};
