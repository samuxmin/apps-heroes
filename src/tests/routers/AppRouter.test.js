import React from "react";
import { mount } from "enzyme";
import { AppRouter } from "../../routers/AppRouter";
import { AuthContext } from "../../auth/AuthContext";

describe("Pruebas en <AppRouter />", () => {
  const contextValue = {
    dispatch: jest.fn(),
    user: {
      logged: false,
    },
  };

  test("debe de mostrar el login si no está autenticado", () => {
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <AppRouter />
      </AuthContext.Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de mostrar el componente de marvel si está autenticado", () => {
    const contextValue = {
      dispatch: jest.fn(),
      user: {
        name: "Sam",
        logged: true,
      },
    };
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <AppRouter />
      </AuthContext.Provider>
    );
    expect(wrapper.find('h1').text()).toBe('Heroes')
  });
});
