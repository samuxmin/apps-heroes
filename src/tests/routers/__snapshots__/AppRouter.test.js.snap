// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter /> debe de mostrar el login si no está autenticado 1`] = `
<div>
  <div
    className="app"
  >
    <h1>
      Login Screen
    </h1>
    <hr />
    <label
      htmlFor="username"
    >
      <b>
        Username
      </b>
    </label>
    <input
      autoComplete="off"
      className="login-input"
      name="username"
      onChange={[Function]}
      placeholder="Username here..."
      type="text"
      value=""
    />
    <br />
    <label
      htmlFor="password"
    >
      <b>
        Password
      </b>
    </label>
    <input
      className="login-input"
      name="password"
      onChange={[Function]}
      placeholder="Password here..."
      type="password"
      value=""
    />
    <p>
      (Cualquier contraseña es valida)
    </p>
    <button
      className="btn btn-login"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
